azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/redux/middleware/AudioMiddleware.kt

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.redux.middleware
+
+import android.media.AudioManager
+import com.azure.android.communication.ui.redux.Dispatch
+import com.azure.android.communication.ui.redux.Middleware
+import com.azure.android.communication.ui.redux.Store
+import com.azure.android.communication.ui.redux.action.Action
+import com.azure.android.communication.ui.redux.action.LocalParticipantAction
+import com.azure.android.communication.ui.redux.state.AudioDeviceSelectionStatus
+import com.azure.android.communication.ui.redux.state.ReduxState
+
+internal interface AudioMiddleware
+
+// This class is middleware that handles Redux requests for AudioDevices
+// You can omit AudioManager in tests and it should behave "as if" it's switching
+internal class AudioMiddlewareImpl(
+    private val audioManager: AudioManager?
```

Comment:
Adam, as per our previous discussions while designing redux, we decided not not inject any android specific class into middleware. would it be possible to have discussion?